### DISCUSSION TITLE
DAQ1301_0 in Artisan

### DISCUSSION THREAD

**Discussion by ivanroaster** [artisan-roaster-scope/artisan#1139] [Q&A] [answered]
Hello, i am trying to pick up the input signal of a DAQ1301 (16x isolated digital input phidget) connected to a wireless hub HUB5000.
Is it supported in Artisan? I am using the extra devices tab and adding it there but not sure what device I should pick from the dropdown list. In y1(x) replacement variable I set it to Y5 and checked LCD1 for display.
In the phidget control panel I can see the state of channel 0 change from true to false when the actuator operates but in the LCD1 it is not showing the signal state, it shows "u.u". The hub is accessed as network/remote.
Any suggestions to solve this?
Thank you

**Comment by bigl00z3** [comment]
I don't see the DAQ1301 on the [Artisan Phidget page](https://artisan-scope.org/devices/phidgets/) so it may not work (yet) with this one (probably need an other user/dev to confirm this).

I imagine that you have checked the Artisan Phidget page and checked how the other DAQs are configured. 

Also to see if your configuration seems OK or not, it is generally a good idea to upload the settings here. It is much easier for others to evaluate your question and answer properly.

_(to attach the settings, don't reply to the email attach it from the GitHub page: save a copy from Artisan, rename it as a TXT extension file, and drop it in a reply box here)._

Nico

**Reply by ivanroaster** [reply]
Hi Nico
Please find the settings page of the input devices next:
Here I was trying with "Phidget HUB IO 23" but I tried with other devices as well with no success.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/129345914/229963736-5526751d-de09-46b7-8351-d04787930656.png">
If DAQ1301 is not supported, then it makes sense I cannot make it work I guess, so the first question I would have is, is it supported/tested? and if so, how should we go about calling it from the extra devices tab?
Thank you
Ivan

**Comment by bigl00z3** [comment]
The DAQ1301 is a Phidget "object" type:
![image](https://user-images.githubusercontent.com/8886922/230031956-d8ba60ec-55cf-4b22-aa0a-92e3c898327d.png)

Objects are more advanced Phidgets and are returning more complex data.  When looking in the Phidget control panel, you don't see a raw signal but it is recognized as the correct object in order to interpret the signal correctly. This means it is is read differently than just an IO value. You would need to call the specific device DAQ1301 which is not currently in the Artisan list.

For it to work, I suspect that it would need to be integrated in a future version of Artisan. As therey are only limited resources, it is then up to the developpers to decide based on their available time, resources and priority. 

Nico

**Reply by ivanroaster** [reply]
Thank you Nico for the explanation, BR, Ivan

**Comment by EspressoMachineWhisperer** [comment]
Hi Ivan. Have you tried selecting Phidget IO Digital 01 in the device selection. I use several phidget inputs and outputs, and i think some are unlisted as modules but available as IO. Make sure that the phidget library is up to date too.

**Reply by MAKOMO** [reply]
Not true. The "Phidget IO Digital" device types work only with the (old USB) Phidget 1010, 1013, 1018 and 1019 devices.

**Comment by zihyer** [comment]
Hey Ivan.  Did you ever get  your DAQ1301_0 to work?  I ordered one and am waiting on it to arrive.  I'm  hoping I can use some of the ideas and advice here to get it working but was wondering if you found a way before I brow beat myself trying to figure it out.  

Thanks much!

**Comment by MAKOMO** [comment]
Added support for Phidgets module DAQ1200, DAQ1300, and the first 8 channels of the DAQ1301 (Commit #65ec7460dfc169ac96f4c12e11702620a7553c66).

Will be available with the next Artisan release to be expected later this month, and within some minutes in the continuous builds.

**Reply by zihyer** [reply]
Thanks Makomo!  I will absolutely be donating to the cause.

**Reply by zihyer** [reply]
Hi Marko,  I have 2.10.0  (c44a180) up and running.  Should I see DAQ1301 explicitly listed under Device Assignment -> Phidget(tab)?  It currently reads: "1010/1011/1013/1018/1019/HUB0000/SBC/DAQ1000/DAQ1400/VCP100x IO"

PS: As promised, signed up for plus!  Whoooot! Thanks for all you and your team does for us.

**Reply by MAKOMO** [reply]
The DAQ is a Phidget IO device available as extra device type and is configured along those other Phidget IO device in the section you quote. Just that section label was not updated. Will do for the next release.

**Comment by EspressoMachineWhisperer** [comment]
Hi Marko.

Fair enough. We have that many running now it’s difficult to remember which is which.

 

Can I confirm with you Marko that 10 is the limit for extra devices?

 

We’ve run out of space for our test roaster that all.

I am also about to start using artisan to chart Espresso extractions, do you know of anyone using Artisan for this purpose as well?

 

Best Regards

Jake Bush

0488130160

Bush-and-Bush-Logo_Landscape signature

**Comment by MAKOMO** [comment]
> Can I confirm with you Marko that 10 is the limit for extra devices? 

yes

> We’ve run out of space for our test roaster that all. I am also about to start using artisan to chart Espresso extractions, do you know of anyone using Artisan for this purpose as well? 

Was done by Londinium on developing their first lever machines. Here is a [post by Frans from 2013](https://artisan-roasterscope.blogspot.com/2013/06/artisan-monitoring-londinium.html) doing similar things.

**Comment by zihyer** [comment]
Ok, I'm stumped.  I figured I would keep my question here since it's directly related to the OP.

Currently, I'm USB connected from my laptop to a VINT Hub00001_0 and from there I have my DAQ1301_0 connected to port 2.  Ports 0 and 1 are BT and ET thermocouples via isolated [TMP1100_0 ](https://www.phidgets.com/?prodid=725) Phidgets respectively on the VINT HUB.  I have a simple switch  ([HIN4208_0](https://www.phidgets.com/?prodid=638) in SPST mode) connected to port 0 of the DAQ1301.  This switch will ultimately be the charge switch connected to the charge level and another like it will be rigged to operate in the same manner for the dump event.  

The intent here is to automagically trigger the "Charge" event in Artisan to begin roast logging and then to end it with a subsequent "Drop" event of a 2nd HIN4208 switch.

I've gone through the "More Phidgets" blog post and that got me up to speed on adding the DAQ1301 01 as an additional device (for ports 0 and 1) but that's where I'm stumped.  How do I go about reading the switch close event from the DAQ, Port0 (and then from port 1 for the Dump switch) and triggering the Charge event by "pressing" the Charge button and start logging?  

I tried loading several commercial machine configs to try to find this event already configured so I could copy it.  I've tried setting the "Change" and "Range" settings a few different ways under the bottom section (for port 0 out of 0-7) in Devices->Phidgets(tab) to no avail.

As a follow on question, can I leave the Charge and Drop buttons active in Artisan in case either of the switches fail to trigger their respective events through a failure of some sort or do the buttons need to be removed from the interface entirely once they are connected to an input?

Many thanks for any help.

**Reply by zihyer** [reply]
Couple photos of my PhrankenRoaster project.  She'll go to the powder coater eventually but I've got a ton of work still to do.
![IMG_3457](https://github.com/artisan-roaster-scope/artisan/assets/110182473/7d1db24b-89b2-4279-9f44-9b1b99d70ea1)
![IMG_3458](https://github.com/artisan-roaster-scope/artisan/assets/110182473/2eacd55d-29dd-4acf-9f02-e914a8b5ef31)

**Comment by MAKOMO** [comment]
You add an extra device to read the signal of your two switches from port 0 and port 1 of your DAQ1301. They report 0 if the switch is open and 1 if it is closed as far as I know. No need to show those values as LCDs or draw their curves.

<img width="1044" alt="Screenshot 2023-12-03 at 12 18 17" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/dca5c5d9-43f5-41b9-b573-bccf548194cf">

Then you add alarm rules to trigger the CHARGE and DROP event. You want to guard such rules by rules to establish the baseline and only get them triggered on state change. So you have a first rules that fires once the first switch reports a 0, but has no action. The second rule is only fired once the first rule is fired (thus the switch was before at 0), guarded by the `If Alarm` reference, and the switch (now) reports 1. This second rules establishes the CHARGE event. You do the same for the other switch to fire the DROP event. You can leave autoDROP/autoCHARGE and the CHARGE/DROP button active. What ever fires the CHARGE/DROP event first will win. The CHARGE/DROP events are never set twice.

<img width="1028" alt="Screenshot 2023-12-03 at 12 27 06" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/46852c6f-7353-4061-8c0d-a2255addbfe7">

**Reply by zihyer** [reply]
I love you.  Seriously,... thank you.  Should you ever find yourself in the Southern Az region of the US, hit us up for some Arizona hospitality!

I can only test one at a time bc I'm waiting on additional cables to connect everything at once but that worked great. Working on the cam lobes and brackets to get things going on the physical side and the next step will be to connect the [Phidget 1141](https://www.phidgets.com/?prodid=114) 15-115kPa gas pressure sensor so I can add a data stream for power.

![DumpSwitch3](https://github.com/artisan-roaster-scope/artisan/assets/110182473/f47c3f03-7bda-49fe-9ca5-cf02df32f345)

**Reply by zihyer** [reply]
Charge switch and thermocouples are up and working fine but the Dump event is being stubborn.  I confirmed switch is working using the Phidget Control Panel app and I even get the "Alarm 4 Triggered" status at the top left in Artisan but it won't seem to trigger the drop event.  I tried checking to see if I had to go through all the buttons (FC end, SC Start,..) in case it was sequential but ofc, it's not.

Manually pressing the Drop button in Artisan does indeed end the roast. 

At "Start" I get this:
![Screenshot 2023-12-05 155245](https://github.com/artisan-roaster-scope/artisan/assets/110182473/a47217fa-1f71-4973-bc9f-523670575f4c)

Then after the Charge lever is opened, I get this: 
![Screenshot 2023-12-05 155347](https://github.com/artisan-roaster-scope/artisan/assets/110182473/1e241a15-f87d-451e-9893-ec5d38fdffa8)

Then when the Dump switch is activated, I get this but the roast continues:
![Screenshot 2023-12-05 155423](https://github.com/artisan-roaster-scope/artisan/assets/110182473/4e904d4a-05b6-4f62-98f6-bc896b18d6f7)

Here are my settings:
![Screenshot 2023-12-05 154920](https://github.com/artisan-roaster-scope/artisan/assets/110182473/979fb4e0-e353-4b00-8ebd-55070ed040cb)
![Screenshot 2023-12-05 155001](https://github.com/artisan-roaster-scope/artisan/assets/110182473/6f223363-1e1e-4dac-9043-413ea050d960)

Any ideas on where I may have gone wrong?

**Reply by MAKOMO** [reply, marked as answer]
You hit a regression :(

The DROP action did not get executed correctly. Fixed in the current continuous build: https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

**Reply by zihyer** [reply]
Well I was a software tester for several years before moving on to dev/sustainment management so it follows reason that I would sniff out one of your only bugs. :D

**Reply by zihyer** [reply]
Marko, just installed [2.10.1](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) [307b823] and I can confirm it fixed the alarm Drop alarm action.  Thank you!

**Reply by MAKOMO** [reply]
Thanks for this confirmation!

**Comment by EspressoMachineWhisperer** [comment]
I am sure Marko will know straight away, but I had a similar problem setting
up mine and found that I had to adjust the scan time setting. The software
alarm was triggered but it was too quick for the output to be fired.

 

Best Regards

Jake Bush

0488130160

Bush-and-Bush-Logo_Landscape signature

**Reply by zihyer** [reply]
That certainly sounds plausible Jake, thanks for the idea.  I searched around and didn't see a scan setting but I was reading through the manual and saw the bit about adding commands to the description field for alarms.  As such, I tried adding "Drop" to alarm 4 to no avail.  I'll keep poking around for scan settings.

**Comment by zihyer** [comment]
Non-sequitur here: I think, for housekeeping purposes, we can safely mark this one as "Answered/Closed".
